src/core/Sampler/Sampler.cpp

```diff
@@ -257,6 +280,30 @@ bool Sampler::renderNote( Note* pNote, unsigned nBufferSize, Song* pSong )
 		return 1;
 	}
 
+	// new instrument and note pan interaction--------------------------
+	// note pan moves the pan in a smaller pan range centered at instrument pan
+
+	// get the note and instrument pan parameters in [-1,1]
+	// TODO a new song member, set in preferences, must point the desired pan parameter Sampler member function
+	float (*getPan)( float, float );
+	// use "ratio" pan parameter (due to the current pan law)
+    getPan = &this->getRatioPan;
+	float fNotePan = getPan( pNote->get_pan_l(), pNote->get_pan_r() );
+	float fInstrPan = getPan( pInstr->get_pan_l(), pInstr->get_pan_r() );
+	
+	// resultant pan: note pan moves the pan in a smaller pan range centered at instrument pan
+	float fPan = fInstrPan + fNotePan * ( 1 - fabs( fInstrPan ) );
+	
+	// Pass fResPan to the Pan Law
+	// use Straight pol pan law.
+	// TODO a new song member, set in preferences, must point the desired pan law Sampler member function		
+	float (*panLaw)( float );
+    panLaw = &this->ratioStraightPolPanLaw;
```

Comment:
How about making a single `Sampler::panLaw( float, enum)` and an enum class for the different pan laws? Inside this function you could use switch-case statement to distinct between the laws.

Reply:
I preferred to use pointers to bypass the conditions checks and to go straight to the function

Reply:
As the addresses of methods depend on the `Sampler` object, maybe this pointer should be saved in some form in the song (for example written as an `int` or `enum`, which type is new to me now), but should live in `Sample` class.
What do you think?

Reply:
> I preferred to use pointers to bypass the conditions checks and to go straight to the function

I see. But I would like to ask you to use switch-case or if statements nevertheless. It's not because I have a personal reference but the code base of Hydrogen does use the former over function pointers. It's more easy to maintain it if there is a consistent coding style. (Admittedly, the one we deal with is not the best but, well, it works. :wink: )

Reply:
I understand the reason.
I would just ask you if you would consider consistent this alternative:
`float (*m_pPanLaw)( float )` is a member in `Sampler`
While import the song, or setting the preferences, the corresponding member function is assigned to `m_pPanLaw` with `if `statements instead of that weird array of addresses